drivers/gles3/shaders/scene.glsl

```diff
@@ -1096,9 +1096,15 @@ void main() {
 #if defined(CUSTOM_IRRADIANCE_USED)
 	ambient_light = mix(ambient_light, custom_irradiance.rgb, custom_irradiance.a);
 #endif // CUSTOM_IRRADIANCE_USED
-	ambient_light *= albedo.rgb;
 
-	ambient_light *= ao;
+	{
+#if defined(AMBIENT_LIGHT_DISABLED)
+		ambient_light = vec3(0.0, 0.0, 0.0);
```

Comment:
I'm not too familiar with this code but it seems a bit wasteful *not to* skip all the `ambient_light` calculations done before if in the end we just set it to 0.

But wait for @clayjohn's more qualified feedback :)

Reply:
This is fine, the shader compiler eliminates all dead code quite rigorously, so anything writing to ambient light before this will be eliminated at compile time.